tests/aggregation/tests.py

```diff
@@ -15,6 +15,76 @@
+        cls.a2 = Author.objects.create(name='test2', age=60)
+        cls.p1 = Publisher.objects.create(name='Apress', num_awards=3, duration=datetime.timedelta(days=1))
+
+        cls.b1 = Book.objects.create(
+            isbn='159059725', name='The Definitive Guide to Django: Web Development Done Right',
+            pages=447, rating=4.5, price=Decimal('30.00'), contact=cls.a1, publisher=cls.p1,
+            pubdate=datetime.date(2007, 12, 6)
+        )
+
+        cls.b2 = Book.objects.create(
+            isbn='067232959', name='Sams Teach Yourself Django in 24 Hours',
+            pages=528, rating=3.0, price=Decimal('23.09'), contact=cls.a2, publisher=cls.p1,
+            pubdate=datetime.date(2008, 3, 3)
+        )
+
+        cls.a1.friends.add(cls.a2)
+        cls.b1.authors.add(cls.a1)
+        cls.b2.authors.add(cls.a2)
+
+    def test_filtered_aggregates(self):
+        agg = Sum('age', filter=Q(name='test2'))
+        self.assertEqual(Author.objects.aggregate(age=agg)['age'], 60)
+
+    def test_double_filtered_aggregates(self):
+        agg = Sum('age', filter=Q(Q(name='test2') & ~Q(name='test')))
+        self.assertEqual(Author.objects.aggregate(age=agg)['age'], 60)
+
+    def test_excluded_aggregates(self):
+        agg = Sum('age', filter=~Q(name='test2'))
+        self.assertEqual(Author.objects.aggregate(age=agg)['age'], 40)
+
+    def test_related_aggregates_m2m(self):
+        agg = Sum('friends__age', filter=Q(friends__name='test2'))
+        self.assertEqual(Author.objects.filter(name='test').aggregate(age=agg)['age'], 60)
+
+    def test_related_aggregates_m2m_and_fk(self):
+        agg = Sum('friends__book__pages',
+                  filter=Q(friends__name='test2', friends__book__publisher__name='Apress'))
+        self.assertEqual(Author.objects.filter(name='test').aggregate(pages=agg)['pages'], 528)
+
+    def test_plain_annotate(self):
+        agg = Sum('book__pages', filter=Q(book__rating__gt=3))
+        qs = Author.objects.annotate(pages=agg).order_by('pk')
+        self.assertSequenceEqual([a.pages for a in qs], [447, None])
+
+    def test_filtered_aggregate_on_annotate(self):
+        pages_annotate = Sum('book__pages', filter=Q(book__rating__gt=3))
+        age_agg = Sum('age', filter=Q(total_pages__gte=400))
+        self.assertEqual(Author.objects.annotate(total_pages=pages_annotate)
+                         .aggregate(summed_age=age_agg), {'summed_age': 40})
+
+    def test_case_aggregate(self):
+        agg = Sum(Case(When(friends__age=40, then=F('friends__age'))),
+                  filter=Q(friends__age=40))
+        self.assertEqual(Author.objects.aggregate(age=agg)['age'], 40)
+
+    def test_aggregate_str(self):
+        q = Q(name='test')
+        agg = Sum('test', filter=q)
+        self.assertIn(str(q), str(agg))
+
+    def test_sum_star_exception(self):
+        with self.assertRaisesMessage(ValueError, 'Star cannot be used with filter. Please specify a field.'):
```

Comment:
You may want to use two lines instead:
    
{{{
    msg = 'Star cannot be used with filter. Please specify a field.'
    with self.assertRaisesMessage(ValueError, msg):
}}}